src/services/websocket_server/handlers/notify_subscribe.rs

```diff
@@ -133,41 +132,41 @@ pub async fn handle(msg: PublishedMessage, state: &AppState) -> Result<()> {
         (account, domain)
     };
 
-    let secret = StaticSecret::random_from_rng(chacha20poly1305::aead::OsRng);
-
-    // Technically we don't need to derive based on client_public_key anymore; we just need a symkey. But this is technical
-    // debt from when clients derived the same symkey on their end via Diffie-Hellman. But now they use the value from
-    // watch subscriptions.
-    let notify_key = derive_key(&client_public_key, &secret)?;
-
     let scope = parse_scope(&request_auth.scp)?;
 
-    let notify_topic = topic_from_key(&notify_key);
-
-    let project_id = project.project_id;
-    info!(
-        "Registering account: {account} with topic: {notify_topic} at project: {project_id}. \
-         Scope: {scope:?}. RPC ID: {id:?}",
-    );
-
-    info!("Timing: Upserting subscriber");
-    let subscriber = upsert_subscriber(
-        project.id,
-        account.clone(),
-        scope.clone(),
-        &notify_key,
-        notify_topic.clone(),
-        &state.postgres,
-        state.metrics.as_ref(),
-    )
-    .await?;
+    let subscriber = {
+        // Technically we don't need to derive based on client_public_key anymore; we just need a symkey. But this is technical
+        // debt from when clients derived the same symkey on their end via Diffie-Hellman. But now they use the value from
+        // watch subscriptions.
+        let secret = StaticSecret::random_from_rng(chacha20poly1305::aead::OsRng);
+        let notify_key = derive_key(&client_public_key, &secret)?;
+        let notify_topic = topic_from_key(&notify_key);
+
+        info!("Timing: Upserting subscriber");
+        upsert_subscriber(
+            project.id,
+            account.clone(),
+            scope.clone(),
+            &notify_key,
+            notify_topic,
+            &state.postgres,
+            state.metrics.as_ref(),
+        )
+        .await?
+    };
     info!("Timing: Finished upserting subscriber");
 
+    let notify_topic = subscriber.topic;
+
     // TODO do in same transaction as upsert_subscriber()
     state
         .notify_webhook(
-            project_id.as_ref(),
+            project.project_id.as_ref(),
+            // if subscriber.inserted {
```

Comment:
Is this comment should be uncommented?

Reply:
No because `WebhookNotificationEvent::Updated` doesn't exist yet. The webhooks are a noop right now, so when I get the webhooks working again then this can be uncommented.

Reply:
Thanks for clarifying this.